File: source-connector/src/main/java/sourceconnector/domain/pipeline/factory/PipelineSupplier.java
package sourceconnector.domain.pipeline.factory;

import sourceconnector.domain.file.FileKey;
import sourceconnector.domain.pipeline.Pipeline;

/**
 * Provides simple factory API to create {@link Pipeline} instance. <br>
 * Pipeline should be created per file handled by the {@link sourceconnector.domain.connect.Task}
 */
@FunctionalInterface
public interface PipelineSupplier<T> {
  Pipeline<T> get(FileKey fileKey);
}

File: source-connector/src/main/kotlin/sourceconnector/domain/pipeline/factory/PipelineSupplier.kt
package sourceconnector.domain.pipeline.factory

import offsetmanager.domain.file.FileKey
import sourceconnector.domain.pipeline.Pipeline

/**
 * Provides simple factory API to create [Pipeline] instance. <br></br>
 * Pipeline should be created per file handled by the [sourceconnector.domain.connect.Task]
 */
fun interface PipelineSupplier<T> {
  fun get(fileKey: FileKey): Pipeline<T?>
}
